### DISCUSSION TITLE
Sequence of instructions in cmd shell doesn't work

### DISCUSSION THREAD

**Discussion by pmatos** [community/community#25569] [Actions]
<p>I have a sequence of commands in a single step like this:</p>
<pre><code>- name: Build
  working-directory: .\racket\src\worksp
  shell: cmd
  run: |
    msvcprep.bat x86_amd64
    build.bat
    ..\..\racket.exe csbuild.rkt
    ..\..\raco.exe pkg install racket-lib
</code></pre>
<p>However, this doesn’t seem to run properly since only the first step executes. All the others are ignored and the next step is executed.</p>
<p>How can I get this to execute all the commands inside the run section?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Thank you for reporting this issue. This is an expected behavior.</p>
<p><em>In ancient DOS versions it was not possible to recursively execute batch files. Then the call command was introduced that called another cmd shell to execute the batch file and returned execution back to the calling cmd shell when finished.</em></p>
<p>You can use call command before bat file name to</p>
<pre><code>call msvcprep.bat x86_amd64
call build.bat
</code></pre>
